feat: log initial connection to the network, printing if client

joshuef opened the pull request.

joshuef committed:
feat: log initial connection to the network, printing if client

joshuef committed:
feat: log build git commit.

This should make it clearer if we're ever using the wrong build somewhere

joshuef committed:
docs: fix typo

RolandSherwin reviewed (approved):
Printing the commit hash would prevent some confusions, atleast for me!

maqi commented on safenode/src/client/api.rs:
```diff
@@ -91,23 +92,31 @@ impl Client {
             match event {
                 ClientEvent::ConnectedToNetwork => {
                     info!("Client connected to the Network.");
+                    println!("Client successfully connected to the Network.");
```
user will only see this `println` , so actually only need to check the `logged_connected` flag here?
then don't need to pass the flag down and do other changes?

joshuef replied:
Ah yeh, clients already did it only once

maqi commented on safenode/build.rs:
```diff
@@ -5,9 +5,26 @@
+    let mut file = File::create("src/git_hash.rs").expect("Failed to create git_hash.rs");
+    writeln!(
+        file,
+        "/// The git commit at build time\npub const GIT_HASH: &str = \"{}\";",
```
shall the detailed `build time` (i.e. `Instant::now()` in readable format) got written as well ?

joshuef replied:
I guess that could be added if you think it's useful?

joshuef replied:
I'm not sure it is? Does it matter if we have the commit? Gonna leave that off, we can always add it in later if desired 👍

joshuef committed:
chore: clients actually dont ned bool update to log connection only once

joshuef committed:
fix: remove git_hash from tracking

joshuef committed:
fix: pull git commit from env var

Sets the env var from build.rs.

This means the git_hash file exists and is commited so tooling
(cargo fmt) won't complain there, and we dont need to run build.rs
just to fmt/check etc
